plugins/modules/aci_access_span_src_group_src.py

```diff
@@ -0,0 +1,442 @@
+    source = module.params.get("source")
+    direction = module.params.get("direction")
+    filter_group = module.params.get("filter_group")
+    drop_packets = module.params.get("drop_packets")
+    epg = module.params.get("epg")
+    routed_outside = module.params.get("routed_outside")
+    state = module.params.get("state")
+    name_alias = module.params.get("name_alias")
+
+    if filter_group and drop_packets:
+        module.fail_json(msg="It is not allowed to configure 'drop_packets: true' when 'filter_group' is configured on the source.")
+    elif epg and drop_packets:
+        module.fail_json(msg="It is not allowed to configure 'drop_packets: true' when 'epg' is configured on the source.")
+    elif routed_outside and drop_packets:
+        module.fail_json(msg="It is not allowed to configure 'drop_packets: true' when 'routed_outside' is configured on the source.")
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="infra",
+            aci_rn="infra",
+        ),
+        subclass_1=dict(
+            aci_class="spanSrcGrp",
+            aci_rn="srcgrp-{0}".format(source_group),
+            module_object=source_group,
+            target_filter={"name": source_group},
+        ),
+        subclass_2=dict(
+            aci_class="spanSrc",
+            aci_rn="src-{0}".format(source),
+            module_object=source,
+            target_filter={"name": source},
+        ),
+        child_classes=["spanRsSrcToFilterGrp", "spanRsSrcToEpg", "spanRsSrcToL3extOut"],
+    )
+
+    aci.get_existing()
+
+    if state == "present":
+        # Create new child configs payload
+        child_configs = []
+        filter_group_tdn = epg_dn = l3ext_out_dn = None
+
+        if filter_group:
+            filter_group_tdn = "uni/infra/filtergrp-{0}".format(filter_group)
+            child_configs.append({"spanRsSrcToFilterGrp": {"attributes": {"tDn": filter_group_tdn}}})
+        if epg:
+            epg_dn = "uni/tn-{0}/ap-{1}/epg-{2}".format(epg.get("tenant"), epg.get("ap"), epg.get("epg"))
+            child_configs.append({"spanRsSrcToEpg": {"attributes": {"tDn": epg_dn}}})
+        elif routed_outside:
+            # encap is set to unknown when not provided to ensure change is executed and detected properly on update
+            encap = "vlan-{0}".format(routed_outside.get("encap")) if routed_outside.get("encap") else "unknown"
+            if routed_outside.get("tenant") and routed_outside.get("l3out"):
+                l3ext_out_dn = "uni/tn-{0}/out-{1}".format(routed_outside.get("tenant"), routed_outside.get("l3out"))
+            else:
+                # tDn is set to "" when not provided to ensure change is executed and detected properly on update
+                l3ext_out_dn = ""
+            child_configs.append({"spanRsSrcToL3extOut": {"attributes": {"encap": encap, "tDn": l3ext_out_dn}}})
+
+        # Validate if existing and remove child objects when do not match provided configuration
```

Comment:
Is this behaviour desirable? See comment above

Reply:
will address in comment above